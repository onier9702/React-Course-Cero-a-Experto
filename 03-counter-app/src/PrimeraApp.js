// FC
import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({name, subtitle}) => {

    // const saludo = `Hello , I am ${name}`;
    return (
        <>
        
            <h1>{name}!!!</h1>;
            <p>{subtitle}</p>;
        </>

    );
};

PrimeraApp.propTypes = {
    name: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
    subtitle: 'I am a subtitle',
}

export default PrimeraApp;