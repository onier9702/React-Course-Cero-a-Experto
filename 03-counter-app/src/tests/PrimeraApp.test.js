import { render } from '@testing-library/react';
import PrimeraApp from  '../PrimeraApp';
import React from 'react'; // Shortcut --->>> imr + Tab
import { shallow } from 'enzyme';

describe('', () => { 

    test('should first show up the message ', () => {
        // --- Antes de Enzyme ----
        // const greeting = 'Hola soy Carlos';
        // const {getByText}  = render(<PrimeraApp name={greeting}/>);
        // expect(getByText(greeting)).toBeInTheDocument();
        // ---- Despues de Enzyme ----
        const nombre = 'Hola soy Carlos';
        const wrapper = shallow(<PrimeraApp name={ nombre }/>);

        expect(wrapper).toMatchSnapshot();
    })

    test('should first show up I am a SUBTITLE', () => { 

        const nombre = 'Hola soy Carlos';
        const subTitle = 'I am a subTitle!!';
        const wrapper = shallow(<PrimeraApp 
            name={ nombre } 
            subtitle= {subTitle}
            />);
        
        const textParagraph = wrapper.find('p').text();
        console.log(textParagraph);
        expect(textParagraph).toBe(subTitle);

     })
 })

