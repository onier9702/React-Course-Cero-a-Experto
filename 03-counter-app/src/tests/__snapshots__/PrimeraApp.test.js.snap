// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should first show up the message  1`] = `
<Fragment>
  <h1>
    Hola soy Carlos
    !!!
  </h1>
  ;
  <p>
    I am a subtitle
  </p>
  ;
</Fragment>
`;
